tests/utils/managed_process.py

```
@@ -631,37 +642,105 @@ def _remove_directory(self, path: str) -> None:
+        """Return at most the last `lines` lines of the log file.
+
+        Streams the file through a bounded deque rather than reading it whole,
+        because a server can emit a very large log before it dies and this runs
+        on the failure path. Returns an empty list if the log is unavailable.
+        """
+        if not self._log_path or not os.path.exists(self._log_path):
+            return []
+        try:
+            with open(self._log_path, "r", encoding="utf-8", errors="ignore") as f:
+                return list(deque(f, maxlen=lines))
+        except Exception as e:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)